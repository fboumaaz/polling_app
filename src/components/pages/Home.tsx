import React, { useState } from "react";
import Grid from "@mui/material/Grid";
import Box from "@mui/material/Box";
import AddQuestionSection from "../molecules/add-question-section/AddQuestionSection";
import VotingSection from "../molecules/voting-section/VotingSection";
import VotingChart from "../molecules/voting-chart/VotingChart";
import optionsProps from "../shared/interfaces/OptionsProps";

const Home = (): JSX.Element => {
  const [options, setOptions] = useState<Array<optionsProps>>([]);
  const [question, setQuestion] = useState("");

  return (
    <Box sx={{ flexGrow: 1, p: 2, my: 4 }}>
      <Grid container rowSpacing={1} columnSpacing={{ xs: 1, sm: 2, md: 3 }}>
        <Grid item xs={4}>
          <AddQuestionSection
            question={question}
            options={options}
            setOptions={(arg: Array<optionsProps>) => setOptions(arg)}
            setQuestion={setQuestion}
          />
        </Grid>
        <Grid item xs={4}>
          <VotingSection
            options={options} question={question}
            setOptions={(arg: Array<optionsProps>) => setOptions(arg)}
          />
        </Grid>
        <Grid item xs={4}>
          <VotingChart options={options} question={question} />
        </Grid>
      </Grid>
     </Box>
  );
};

export default Home;
