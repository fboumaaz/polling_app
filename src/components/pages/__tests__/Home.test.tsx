import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Home from "../Home";

configure({ adapter: new Adapter() });

describe('home page', () => {
  const wrapper = shallow(<Home />);

  it('renders AddQuestionSection component', () => {
    expect(wrapper.find('AddQestionSection')).toHaveLength(1);
  });

  it('renders VotingSection component', () => {
    expect(wrapper.find('VotingSection')).toHaveLength(1);
  });

  it('renders VotingChart component', () => {
    expect(wrapper.find('VotingChart')).toHaveLength(1);
  });

  it('renders snapshots', () => {
    expect(wrapper).toMatchSnapshot()
  });
});
