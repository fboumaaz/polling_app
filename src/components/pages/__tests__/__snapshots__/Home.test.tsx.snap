// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VotingSection component renders snapshots 1`] = `ShallowWrapper {}`;

exports[`home page renders snapshots 1`] = `ShallowWrapper {}`;
